libImaging/Quant.c

```diff
@@ -1251,13 +1271,20 @@ quantize(Pixel *pixelData,
    free_box_tree(root);
    root=NULL;
 
-   qp=malloc(sizeof(uint32_t)*nPixels);
+   /* malloc check ok, using calloc for overflow */
+   qp=calloc(nPixels, sizeof(uint32_t));
    if (!qp) { goto error_4; }
 
-   avgDist=malloc(sizeof(uint32_t)*nPaletteEntries*nPaletteEntries);
+   if ((nPaletteEntries > UINT32_MAX / nPaletteEntries ) ||
+       (nPaletteEntries > UINT32_MAX / (nPaletteEntries * sizeof(uint32_t)))) {
```

Comment:
The nPaletteEntries ^2 < UINT32_MAX is more restrictive than checking against UINT32_MAX /  4. 

The second check does not look necessary, as there's no looping that relies on that n^2*4 value not overflowing, as we're not iterating over it as chars, only ints.